Signal/src/ViewControllers/MenuActionsViewController.swift

```diff
@@ -329,6 +353,47 @@ class MenuActionSheetView: UIView, MenuActionViewDelegate {
         mask.path = path.cgPath
         self.layer.mask = mask
     }
+
+    private func unhighlightAllActionViews() {
+        for actionView in actionViews {
+            actionView.isHighlighted = false
+        }
+    }
+
+    private func actionView(touchedBy touchPoint: CGPoint, fromView: UIView) -> MenuActionView? {
+        for actionView in actionViews {
+            let convertedPoint = actionView.convert(touchPoint, from: fromView)
+            if actionView.point(inside: convertedPoint, with: nil) {
+                return actionView
+            }
+        }
+        return nil
+    }
+
+    private func highlightActionView(location: CGPoint, fromView: UIView) {
+        guard let touchedView = actionView(touchedBy: location, fromView: fromView) else {
+            unhighlightAllActionViews()
+            return
+        }
+
+        if hasEverHighlightedAction, !touchedView.isHighlighted {
+            self.hapticFeedback.selectionChanged()
```

Comment:
So we use haptic feedback if the selection _changes_ but not if you just tap on an item.

Reply:
Yup, exactly. That's consistent with how Apple uses (and specifies) this to be used.